packages/xml-views-validation/src/validators/document/non-unique-id.ts

```diff
@@ -12,62 +12,75 @@ import {
 } from "@ui5-language-assistant/user-facing-text";
 import { NonUniqueIDIssue } from "../../../api";
 import { isPossibleCustomClass } from "../../utils/ui5-classes";
+import { Context } from "@ui5-language-assistant/context";
+import { locationToRange } from "../../utils/range";
 
 const { NON_UNIQUE_ID } = validations;
+type DuplicatedIDXMLAttribute = XMLAttribute & { syntax: { value: XMLToken } };
+interface IDCollector {
+  attribute: DuplicatedIDXMLAttribute;
+  documentPath: string;
+}
 
-export function validateNonUniqueID(xmlDoc: XMLDocument): NonUniqueIDIssue[] {
+export function validateNonUniqueID(
+  xmlDoc: XMLDocument,
+  context: Context
+): NonUniqueIDIssue[] {
   const idCollector = new IdsCollectorVisitor();
-  accept(xmlDoc, idCollector);
+  const files = Object.keys(context.viewFiles);
```

Comment:
Do I understand this correctly that every time we validate a document we go through all the XML templates and collect all the IDs? This will have poor performance scaling, we should only do this at most once per diagnostic cycle. It is probably fine when editing a single file, but on start up with many open files we could see performance degradation.

Reply:
yes. Performance wise did not notice any, but this can be optimized. Now control ids are part of context and reused in validation and quick fix.